### ISSUE TITLE
Have NVIDIA driver and other related dependencies as part of the Linux AMI

### ISSUE THREAD

**Issue by huydhn** [pytorch/pytorch#85296]
### 🚀 The feature, motivation and pitch

At the moment, NVIDIA driver and some other related dependencies like nvidia-docker2 are installed in the CI because they are not part of the base Amazon Linux AMI that we are using [ami-096198a0bccc6bad4](https://github.com/fairinternal/pytorch-gha-infra/blob/main/runners/main.tf#L59).  Specifically, the step is here:

* https://github.com/pytorch/pytorch/blob/master/.github/workflows/_linux-test.yml#L76-L85
* https://github.com/pytorch/pytorch/blob/master/.github/workflows/_binary-test-linux.yml#L173-L184

AFAIK, there is an ongoing task to host these dependencies on S3 to improve the CI reliability https://github.com/pytorch/pytorch/issues/75703.  Nevertheless, it makes sense to eventually have https://github.com/pytorch/pytorch/blob/master/.github/scripts/install_nvidia_utils_linux.sh as part of the Linux AMI we are using for CUDA testing.  As an example, Windows AMI already has them.

cc @seemethere @malfet @pytorch/pytorch-dev-infra @atalman

**Comment by huydhn** [comment]
A brief chat with @atalman:  One possible reason that we don't have this yet is that we use the same base AMI for cuda and non-cuda (cpu).  If that's the case, one action item could be to have a different AMI for cuda testing

**Comment by ZainRizvi** [comment]
@huydhn, is this still an open item?

**Comment by huydhn** [comment]
I think let's close it because the issue is still open, but it's of low priority.  We install the driver on CI, but it happens only once on each runner, and there is no report of flakiness recently.  We don't need to take any further actions here for now.
